pkg/controllers/secretstore/common.go

```diff
@@ -74,6 +78,11 @@ func reconcile(ctx context.Context, req ctrl.Request, ss esapi.GenericStore, cl
 	err := validateStore(ctx, req.Namespace, opts.ControllerClass, ss, cl, opts.GaugeVecGetter, opts.Recorder)
 	if err != nil {
 		log.Error(err, "unable to validate store")
+		// in case of validation status unknown, validateStore will mark
+		// the store as ready but we should show ReasonValidationUnknown
+		if errors.Is(err, validationUnknownError) {
+			return ctrl.Result{RequeueAfter: requeueInterval}, err
```

Comment:
this will never do what this intends to, because requeueAfter is ignored if you return an error . Please see https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/reconcile/reconcile.go#L30-L47 for more information

You should also see a warning because both err and requeue after have been set